ethcore/src/blockchain/blockchain.rs

```diff
@@ -1350,11 +1359,13 @@ impl BlockChain {
 				}
 			},
 			BlockLocation::BranchBecomingCanonChain(ref data) => {
-				let ancestor_number = self.block_number(&data.ancestor).unwrap();
+				let ancestor_number = self.block_number(&data.ancestor)
+					.expect("Inserted block's ancestor number always exists; qed");
 				let start_number = ancestor_number + 1;
 
 				let mut blooms: Vec<Bloom> = data.enacted.iter()
-					.map(|hash| self.block_header_data(hash).unwrap())
+					.map(|hash| self.block_header_data(hash)
+						 .expect("Inserted block's header data always exists; qed"))
```

Comment:
"hash belongs to an inserted block; block header data of an inserted block is always available; qed"